src/DXDecompiler/DX9Shader/Decompiler/RegisterState.cs

```diff
@@ -188,22 +188,25 @@ public string GetSourceName(InstructionToken instruction, int srcIndex)
 							throw new NotImplementedException();
 					}
 					var registerNumber = instruction.GetParamRegisterNumber(srcIndex);
-					ConstantDeclaration decl = FindConstant(parameterType, registerNumber);
+					ConstantDeclaration decl = FindConstant(registerNumber);
 					if(decl == null)
 					{
 						// Constant register not found in def statements nor the constant table
 						//TODO:
 						return $"Error {registerType}{registerNumber}";
 						//throw new NotImplementedException();
 					}
-
-					if(decl.ParameterClass == ParameterClass.MatrixRows)
+					var totalOffset = registerNumber - decl.RegisterIndex;
+					var data = decl.GetRegisterTypeByOffset(totalOffset);
```

Comment:
GetRegisterTypeByOffset will throw an System.NotImplementedException when called on objects (samplers/textures). can either add a check here

```
if(decl.ParameterClass == ParameterClass.Object)
{
	return decl.Name;
}
```
or add ParameterClass.Object handling to GetRegisterTypeByOffset. 

Considering DX9 supports sampler array indexing, does to extend GetRegisterTypeByOffset to support objects?

Reply:
tab size = 4 is fine =)

Reply:
@spacehamster Now sampler array is supported too: https://github.com/spacehamster/DXDecompiler/blob/5fe8c2f1bb8af8d8bceef1255a76d1f7c86342de/src/DXDecompiler.Tests/TestFixtures/TestDX9Effects.cs#L167